Automattic/wp-calypso
# Site Settings: Update Traffic section to use the new settings controller

This PR updates the Traffic section to use the new settings controller that we introduced in #12493. Previously we were duplicating almost the same code for the Traffic section controller, and now we'll be using the same one for all other settings sections that we'll separate from the settings chunk. 

This PR adds no real performance improvement (as the Traffic tab is already a separate chunk), but it demonstrates how we can move a settings tab in its own section. Following this example, we can move out the rest of the settings tabs in their own chunks, which will reduce the size of the main `settings` one.

For background on why we're moving settings sections to separate chunks, you can refer to p4TIVU-5Zs-p2.

To test:
* Checkout this branch.
* Test all Traffic tab cases and verify there are no regressions there: 
  * `/settings/traffic/`
  * `/settings/traffic/$site` for a Jetpack site
  * `/settings/traffic/$site` for a WordPress.com site
* Verify there are no regressions in the settings and toggles in the Traffic tab by playing with them a bit.
